cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/view/AwsNetworkView.java

```diff
@@ -55,6 +57,23 @@ public boolean isSubnetList() {
         return isSubnetList() ? List.of(getExistingSubnet().split(",")) : (isExistingSubnet() ? List.of(getExistingSubnet()) : List.of());
     }
 
+    private String getEndpointGatewaySubnet() {
+        return network.getStringParameter(ENDPOINT_GATEWAY_SUBNET_ID);
+    }
+
+    public boolean isEndpointGatewaySubnetSet() {
```

Comment:
Seeing the name of this function together with the one below, I was just confused why we are having a check for both a `Set` (as in HashSet) and a `List` check. It turns out it is a check for existence. 

Maybe we can merge the List vs non-list cases to simplify. If there is only 1 subnet, then it is a list of 1.

Reply:
I can rename the function to make it more clear. I don't really want to change the methods themselves because they were written following the same pattern as is used for the environment subnet list.

Reply:
Changed this method name to `containsEndpointGatewaySubnet`.